src/Calendar/Grid/GridRow.tsx

```diff
@@ -78,35 +80,49 @@ const GridRow = forwardRef<'div', GridRowProps>((props: GridRowProps, ref) => {
       //           Calendar is not supposed to be reused this way
       let inRange = false;
       // for Selected
-      if (selectedStartDate && selectedEndDate) {
-        if (isBefore(thisDate, selectedEndDate) && isAfter(thisDate, selectedStartDate)) {
+      if (selectedStartDateJS && selectedEndDateJS) {
+        const selectedStartDate = {
+          year: selectedStartDateJS.getFullYear(),
+          month: selectedStartDateJS.getMonth() + 1,
+          day: selectedStartDateJS.getDate()
+        };
+        const selectedEndDate = {
+          year: selectedEndDateJS.getFullYear(),
+          month: selectedEndDateJS.getMonth() + 1,
+          day: selectedEndDateJS.getDate()
+        };
+        if (compare(thisDate, selectedEndDate) < 0 && compare(thisDate, selectedStartDate) > 0) {
           inRange = true;
         }
-        if (isBefore(thisDate, selectedStartDate) && isAfter(thisDate, selectedEndDate)) {
+        if (compare(thisDate, selectedStartDate) < 0 && compare(thisDate, selectedEndDate) > 0) {
           inRange = true;
         }
       }
 
       // for Hovering
-      if (!isSelected && hoverStartDate && hoverEndDate) {
-        if (!isAfter(thisDate, hoverEndDate) && !isBefore(thisDate, hoverStartDate)) {
+      if (!isSelected && hoverStartDateJS && hoverEndDateJS) {
+        const hoverStartDate = {
+          year: hoverStartDateJS.getFullYear(),
+          month: hoverStartDateJS.getMonth() + 1,
+          day: hoverStartDateJS.getDate()
+        };
+        const hoverEndDate = {
+          year: hoverEndDateJS.getFullYear(),
+          month: hoverEndDateJS.getMonth() + 1,
+          day: hoverEndDateJS.getDate()
+        };
+        if (compare(thisDate, hoverEndDate) <= 0 && compare(thisDate, hoverStartDate) >= 0) {
           inRange = true;
         }
-        if (!isAfter(thisDate, hoverStartDate) && !isBefore(thisDate, hoverEndDate)) {
+        if (compare(thisDate, hoverStartDate) <= 0 && compare(thisDate, hoverEndDate) >= 0) {
```

Comment:
<img width="1468" height="719" alt="image" src="https://github.com/user-attachments/assets/dce0725e-0bbc-4487-b451-abb459b6e670" />

The judgment verification of the selected date has functional issues.

Reply:
https://github.com/user-attachments/assets/d582461c-0231-4880-9fd3-840dde6bd180

Reply:
Fixed